src/services/services.ts

```diff
@@ -1126,12 +1126,21 @@ namespace ts {
         return createLanguageServiceSourceFile(sourceFile.fileName, scriptSnapshot, sourceFile.languageVersion, version, /*setNodeParents*/ true, sourceFile.scriptKind);
     }
 
+    const NoopCancellationToken: CancellationToken = {

```

Comment:
This might be a premature optimization.  Happy to roll it back if people have concerns.